### PR TITLE
Upgrade to webpack 4

### PR DESCRIPTION
**Pull request by tomaszpaulus**
@pun-ky please consider testing current state on branch
-  for now js and scss compilation works
- webpack 4 is used
- modular approach to webpack tasks is used

### CODE REVIEW

**Comment by pun-ky** [comment]
Nice changes, of course I'll test it soon. Greetings 😉

File: app/design/package.json [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+    "html-webpack-plugin": "^3.2.0",
+    "mini-css-extract-plugin": "0.4.2",
+    "node-sass": "4.9.3",
+    "optimize-css-assets-webpack-plugin": "^5.0.1",
+    "postcss-focus": "^4.0.0",
+    "postcss-loader": "^3.0.0",
+    "postcss-reporter": "^6.0.0",
+    "sass-loader": "7.1.0",
+    "style-loader": "0.23.0",
+    "stylelint": "^9.5.0",
```
Comment (pun-ky) [inline app/design/package.json]: [3/4] Linking dependencies...
warning "stylelint > postcss-jsx > postcss-styled@0.34.0" has incorrect peer dependency "postcss-syntax@>=0.34.0".

File: app/design/src/main/content/jcr_root/etc/designs/example/publish/src/styles/main.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+@import 'module';
```
Comment (pun-ky) [inline app/design/src/main/content/jcr_root/etc/designs/example/publish/src/styles/main.scss]: Happy[js]: Version: 5.0.0. Threads: 4
Happy[js]: All set; signaling webpack to proceed.
[hardsource:0a693d4f] Writing new cache 0a693d4f...
[hardsource:0a693d4f] Tracking node dependencies with: package.json, node_modules/*.

styles/_module.scss
1:1     ÔÇ╝  Unexpected empty source (no-empty-source) [stylelint]


styles/main.scss
2:23    ÔÇ╝  Expected newline before "}" of a multi-line block (block-closing-brace-newline-before) [stylelint]

Hash: ff054a7621faa9802378
Version: webpack 4.17.2

Comment (pun-ky) [reply app/design/src/main/content/jcr_root/etc/designs/example/publish/src/styles/main.scss]: image

main.css is at bad level. clientlib is referencing file dist/main.css not dist/css/main.css

File: common.gradle [hunk 3]
```diff
@@ -72,8 +72,8 @@ allprojects { subproject ->
 
     plugins.withId 'com.moowork.node', {
         node {
-			version = '6.11.1'
-            yarnVersion = '0.27.5'
+			version = '8.9.0'
```
Comment (pun-ky) [inline common.gradle]: please fix formatting here

Comment (tomaszpaulus) [reply common.gradle]: fixed

**Pull request merged**